Sources/SnapshotTesting/Snapshotting/SwiftUIView.swift

```diff
@@ -37,6 +37,14 @@ extension Snapshotting where Value: SwiftUI.View, Format == UIImage {
     traits: UITraitCollection = .init()
     )
     -> Snapshotting {
+      #if compiler(>=5.7)
+      if #available(iOS 16, *), drawHierarchyInKeyWindow == false {
+        // Use iOS 16 ImageRenderer strategy if available
+        return .imageRenderer(precision: precision, layout: layout, traits: traits)
```

Comment:
Do you think it's safe to automatically opt into this new strategy whenever you build for an iOS 16 simulator vs. not? Building for a different OS typically invalidates existing screen shots, so I'm thinking maybe this is fine.

Reply:
I was just doing some more testing and ran into an issue that would prevent the `ImageRender` approach from being a drop-in replacement for `.image`.

Specifically this warning from the docs:

> Important
> 
> ImageRenderer output only includes views that SwiftUI renders, such as text, images, shapes, and composite views of these types. It does not render views provided by native platform frameworks (AppKit and UIKit) such as web views, media players, and some controls. For these views, ImageRenderer displays a placeholder image, similar to the behavior of [drawingGroup(opaque:colorMode:)](https://developer.apple.com/documentation/swiftui/view/drawinggroup(opaque:colormode:)).

In my test `NavigationView` also falls into that category, so it captures a placeholder image instead of any content in a `NavigationView`.

Would you suggest making a new `.imageRender` type in the library (name up for bike shed-ing), make the behavior opt-in/opt-out, or create a separate "Plug-in" repository?

Reply:
I think `imageRenderer` is solving a common enough need that it's something we'd be willing to take into the repo. Whenever we decide to have a bigger breaking change/redesign, we can consider how plugins work.

Wanna make those changes and let me know when it's ready to merge?